### PR TITLE
fix: update calculate fee outputs

### PR DESCRIPTION
**Pull request by Torres-ssf**
Made `CalculateTransactionFee` to return `minFee` and `maxFee`. 

These values are relevant when calculating the amount of funds a Transaction Request needs before being submitted to the chain

**EDIT**

On [1307](https://github.com/FuelLabs/fuels-ts/pull/1307), I did not return the `maxFee` because I was under the impression that it was not relevant, since the actual fee deducted from the user's wallet was always the `minFee`.

However, I've recently become aware that the [VM always uses](https://github.com/FuelLabs/fuel-vm/blob/94677aa1cfa5a150368c4820a8a757baac479e1f/fuel-vm/src/checked_transaction/balances.rs#L73-L78) the `maxFee` value to estimate if the transaction has enough balances to cover the fee.

In other words, we need `maxFee` in order to properly fund a transaction before submitting it to the chain.

### CODE REVIEW

**Review by arboleya — APPROVED** [review]
The code looks ok, but I must admit, I didn't follow the change.

For example, why is this right and the previous one wrong?

I'm assuming we had no errors before, and both seem to work ok.

> The terminology `feeFromGasUsed` also feels weird, but maybe it's me out of context.

**Comment by Torres-ssf** [comment]
> The code looks ok, but I must admit, I didn't follow the change.
> 
> For example, why is this right and the previous one wrong?
> 
> I'm assuming we had no errors before, and both seem to work ok.
> 
> > The terminology `feeFromGasUsed` also feels weird, but maybe it's me out of context.

@arboleya apologies for the lack of a deeper context here.

On [1307](https://github.com/FuelLabs/fuels-ts/pull/1307), I did not return the `maxFee` because I was under the impression that it was not relevant, since the actual fee deducted from the user's wallet was always the `minFee`.

However, I've recently become aware that the [VM always uses](https://github.com/FuelLabs/fuel-vm/blob/94677aa1cfa5a150368c4820a8a757baac479e1f/fuel-vm/src/checked_transaction/balances.rs#L73-L78) the `maxFee` value to estimate if the transaction has enough balances to cover the fee.

In other words, we need `maxFee` in order to properly fund a transaction before submitting it to the chain.

The value from `feeFromGasUsed` is related to the part of the total fee that will always be charged depending on the amount of gas used.

**Review by danielbate — APPROVED** [review]
